### DISCUSSION TITLE
Why are most instrumentation modules disabled when running in OpenShift?

### DISCUSSION THREAD

**Discussion by m-kay** [open-telemetry/opentelemetry-java-instrumentation#6128] [Q&A] [answered]
I have a very strange behavior that most instrumentation modues are disabled when running the javaagent instrumentation in OpenShift. When I run the same image locally in docker the modules are enabled. For example spring-kafka is disabled and I get the following log:
`[otel.javaagent 2022-06-01 15:22:44:738 +0200] [main] DEBUG io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule - Instrumentation spring-kafka is disabled`

Any idea why this could behave differently?

**Comment by laurit** [comment, marked as answer]
There was a similar report https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/5885 where it turned out to be caused by dynatrace agent.

**Reply by m-kay** [reply]
Thank you for that hint, seems to be the same in my case. 

Do you know if there is a way around that other than completely disable the dynatrace agent?

**Reply by laurit** [reply]
@Oberon00 could you or someone else from Dynatrace answer this.

**Reply by Oberon00** [reply]
From what I know, the Dynatrace agent disables all instrumentations for which it has implementations on its own, if the OpenTelemetry API support is also enabled. If you disable the respective Dynatrace instrumentation (aka sensor) in the Dynatrace Web UI, the OpenTelemetry instrumentation will be left enabled. Usually though, the Dynatrace instrumentations should play nicer with the Dynatrace agent & backend.

**Reply by tinnapat** [reply]
I have the same problem. In my case I had to disable container monitoring for the given namespace in OpenShift, which disable all Dynatrace monitoring include resources and JVM. Is it possible to have Dynatrace disable only tracing with conflict with OpenTelemetry but keep monitoring the resources and JVM?

**Reply by Oberon00** [reply]
Please find the documentation at <https://www.dynatrace.com/support/help/shortlink/opentelemetry-oneagent>
I believe you can disable Dynatrace agent doing anything with OpenTelemetry by going to  Settings > Server-side service monitoring > Deep monitoring, then to Troubleshooting (or OneAgent features) and disable "OpenTelemetry (Java)". Though maybe it would be better to disable just the OneAgent feature that is in conflict with the OTel instrumentation you want to have (probably "Java Kafka" in the original issue).
If you still have trouble, I recommend going through the official Dynatrace support channels.
